Wrappers/Python/cil/optimisation/algorithms/PDHG.py

```diff
@@ -74,41 +252,76 @@ def __init__(self, f=None, g=None, operator=None, tau=None, sigma=1.,initial=Non
+        self.set_gamma_fconj(kwargs.get('gamma_fconj', None))  
+        if self.gamma_g is not None and self.gamma_fconj is not None:
+            raise ValueError("The adaptive update of the PDHG stepsizes in the case where both functions are strongly convex is not implemented at the moment.")                           
 
-    def set_up(self, f, g, operator, tau=None, sigma=1., initial=None):
-        '''initialisation of the algorithm
+        self.set_up(f=f, g=g, operator=operator, tau=tau, sigma=sigma, initial=initial, **kwargs)
 
-        :param operator: a Linear Operator
-        :param f: Convex function with "simple" proximal of its conjugate. 
-        :param g: Convex function with "simple" proximal 
-        :param tau: Step size parameter for Primal problem
-        :param sigma: Step size parameter for Dual problem
-        :param initial: Initial guess ( Default initial = 0)'''
+    @property
+    def tau(self):
+        return self._tau   
 
+    @property
+    def sigma(self):
+        return self._sigma   
+
+    @property
+    def gamma_g(self):
+        return self._gamma_g 
+
+    @property
+    def gamma_fconj(self):
+        return self._gamma_fconj    
+                       
+    def set_gamma_g(self, value):
+        
+        if isinstance (value, Number): 
+            if value <= 0:                      
+                raise ValueError("Strongly convex constant is a positive number, {} is passed for the strongly convex function g.".format(value))                   
+            self._gamma_g = value  
+        elif value is None:
+            pass             
+        else:            
+            raise ValueError("Positive float is expected for the strongly convex constant of function g, {} is passed".format(value))        
+        
+        
+    def set_gamma_fconj(self, value):
+  
+        if isinstance (value, Number): 
+            if value <= 0:                     
+                raise ValueError("Strongly convex constant is positive, {} is passed for the strongly convex conjugate function of f.".format(value))   
+            self._gamma_fconj = value    
+        elif value is None:
+            pass
+        else: 
+            raise ValueError("Positive float is expected for the strongly convex constant of the convex conjugate of function f, {} is passed".format(value))                       
+        
+                  
+    def set_up(self, f, g, operator, tau=None, sigma=None, initial=None, **kwargs):
+        """Initialisation of the algorithm
+        """
         print("{} setting up".format(self.__class__.__name__, ))
         
-        # can't happen with default sigma
-        if sigma is None and tau is None:
-            raise ValueError('Need sigma*tau||K||^2<1')
-        # algorithmic parameters
+        # Triplet (f, g, K)
         self.f = f
         self.g = g
         self.operator = operator
 
-        self.tau = tau
-        self.sigma = sigma
+        #Default step sizes
```

Comment:
this comment should be in the actual method. `set_step_sizes` will use default values if the user didn't pass anything, but it would take different paths when `sigma` and/or `tau` are passed